studiopress/genesis-custom-blocks
# Auto-slug block and field names, make Repeater work

Updated December 16th

#### Tickets
[GF-2518](https://wpengine.atlassian.net/browse/GF-2518): Auto-slug for block name
[GF-2519](https://wpengine.atlassian.net/browse/GF-2519): Auto-slug for field names
[GF-2507](https://wpengine.atlassian.net/browse/GF-2507): Settings for Repeater
[GF-2508](https://wpengine.atlassian.net/browse/GF-2508): Add / Remove Sub-fields
[GF-2509](https://wpengine.atlassian.net/browse/GF-2509): Move / Resize Sub-fields
[GF-2510](https://wpengine.atlassian.net/browse/GF-2510): Sub-field JSON data
[GF-2511](https://wpengine.atlassian.net/browse/GF-2511): Save sub-field data

#### Changes
* Enables Repeater `sub_fields`
* When there is no block name (slug), this sets it when entering the block title:
![there-slugging](https://user-images.githubusercontent.com/4063887/101060845-75379000-3555-11eb-8ed2-6b81f6f2943c.gif)

#### Testing instructions
1. Follow 'Testing Instructions' 1-5 in https://github.com/studiopress/genesis-custom-blocks-pro/pull/16#issue-515954659
1. Test auto-slugging for the block name ([GF-2518](https://wpengine.atlassian.net/browse/GF-2518)):
![adsf-asdsdf](https://user-images.githubusercontent.com/4063887/102325265-91213580-3f48-11eb-8754-6b8a435d2cf1.gif)
1. Expected: if you click somewhere else and edit the title again, it shouldn't auto-slug anymore
1. Test auto-slugging for the field name ([GF-2519](https://wpengine.atlassian.net/browse/GF-2519)):
![ijsdf-=sdf](https://user-images.githubusercontent.com/4063887/102325981-6f747e00-3f49-11eb-894e-20851bcfc3b1.gif)
1. Expected: if you click somewhere else and edit the Field Label again, it shouldn't auto-slug anymore
1. Change a field to 'Repeater'
1. Expected: these 3 settings appear ([GF-2507](https://wpengine.atlassian.net/browse/GF-2507)):
![isdfoij](https://user-images.githubusercontent.com/4063887/102326682-50c2b700-3f4a-11eb-92d9-355d2d1fe1db.png)
1. Test adding and deleting Repeater sub_fields ([GF-2508](https://wpengine.atlassian.net/browse/GF-2508)): ![iojsdfsf](https://user-images.githubusercontent.com/4063887/102327225-0a218c80-3f4b-11eb-970c-abbca9ad8532.gif)
1. Test moving and resizing Repeater sub_fields ([GF-2509](https://wpengine.atlassian.net/browse/GF-2509)):
![in-drig](https://user-images.githubusercontent.com/4063887/102327927-05110d00-3f4c-11eb-8323-f6009c34f545.gif)
1. Try to break any of the above 😄
